cleverpush/src/main/java/com/cleverpush/CleverPush.java

```diff
@@ -571,6 +575,21 @@ public void initFeatures() {
             this.pendingInitFeaturesCall = true;
             return;
         }
+        IntentFilter intentFilter = new IntentFilter(Constants.DEVICE_ID_INTENT);
+        if (intentFilter != null) {
+            context.registerReceiver(deviceIdBroadcastReceiver, intentFilter);
+        }
+
+        SharedPreferences sharedPreferences = getSharedPreferences(getContext());
+        if (sharedPreferences.getString(CleverPushPreferences.DEVICE_ID, null) == null) {
+            sharedPreferences.edit().putString(CleverPushPreferences.DEVICE_ID, UUID.randomUUID().toString()).apply();
+        }
+
+        this.getChannelConfig(channelConfig1 -> {
+            if (channelConfig.optString(Constants.DEVICE_ID_CONFIG_FIELD) != null && channelConfig.optBoolean(Constants.DEVICE_ID_CONFIG_FIELD) == true) {
+                sendBrodcastReceiver();
```

Comment:
arent we overwriting the deviceID of the first App like this e.g.
We install App1 have deviceId X
We install App2 have deviceId Y, sendBroadcast, App1 recieves broadcast and changes deviceId from X to Y
But we want all Apps to have deviceId of App1

Reply:
123

Reply:
123